sats-dna/src/main/kotlin/com/sats/dna/components/SatsSearchBar.kt

```diff
@@ -0,0 +1,179 @@
+package com.sats.dna.components
+
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.text.BasicTextField
+import androidx.compose.foundation.text.KeyboardActions
+import androidx.compose.foundation.text.KeyboardOptions
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.SolidColor
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardCapitalization
+import androidx.compose.ui.unit.dp
+import com.sats.dna.components.TrailingContent.Icon
+import com.sats.dna.components.button.SatsButtonColor
+import com.sats.dna.components.button.SatsIconButton
+import com.sats.dna.theme.SatsTheme
+import com.sats.dna.tooling.LightDarkPreview
+
+@Composable
+fun M3SatsSearchBar(
+    query: String,
+    onQueryChange: (query: String) -> Unit,
+    onClearClicked: () -> Unit,
+    modifier: Modifier = Modifier,
+    enabled: Boolean = true,
+    keyboardActions: KeyboardActions = KeyboardActions.Default,
+    placeholder: @Composable (() -> Unit)? = null,
```

Comment:
Maybe it's cleaner to just allow a `String?` for the placeholder? Then we create the `Text()` internally with the correct styles. Alternatively, we can wrap the usages of the `placeholder: @Composable () -> Unit` inside of a `ProvideTextStyle {}`.

Reply:
True